### PR TITLE
Replace private CCCryptorGCMOneshotEncrypt with CryptoKit AES.GCM (v3.0.0)

### PR DESCRIPTION
**Pull request by aleks-tpom6oh**
## Summary

- **Fixes App Store rejection** by removing the private `CCCryptorGCMOneshotEncrypt` symbol from `AvoEncryption.m`
- Replaces it with a thin Swift wrapper (`AvoGCMEncryptor`) around CryptoKit `AES.GCM.seal()`
- Introduces **wire format version 0x01** with 12-byte nonces (CryptoKit constraint) instead of the previous 16-byte IVs (version 0x00)
- Bumps SDK to **v3.0.0** (breaking: deployment target iOS 12 → iOS 13)
- Removes ~200 lines of custom big-number arithmetic for compressed key decompression, replaced by CryptoKit `P256.KeyAgreement.PublicKey(compressedRepresentation:)` (iOS 16+) with Security framework fallback (iOS 13-15)

## Wire Format Change

| Field | v0x00 (legacy) | v0x01 (this PR) |
|-------|----------------|-----------------|
| Version byte | `0x00` | `0x01` |
| Nonce/IV length | 16 bytes | 12 bytes |
| Header size | 98 bytes | 94 bytes |
| Layout | `[Ver(1)][PubKey(65)][IV(16)][Tag(16)][Ciphertext]` | `[Ver(1)][PubKey(65)][Nonce(12)][Tag(16)][Ciphertext]` |

## Deployment Order

> **The monorepo web app must deploy first** to accept both v0x00 and v0x01 payloads. The server-side changes (ReScript `CryptoHelper.res` + `CryptoUtil.res`) are in the companion monorepo PR: https://github.com/avohq/monorepo/pull/9034
>
> **Testing:** To verify end-to-end decryption, the tester should run the web app from the monorepo PR branch locally.

### 1. CocoaPods Infrastructure
- `AvoInspector.podspec` — v3.0.0, iOS 13.0, Swift 5.0, CryptoKit framework
- `Example/Podfile` — iOS 13.0 deployment target
- `AvoInspector.m` — libVersion "3.0.0"

### 2. CryptoKit Swift Wrapper (new files)
- `AvoGCMEncryptor.swift` — thin CryptoKit wrapper (encrypt + key decompression)
- `AvoGCMEncryptor.h` — manual ObjC header (avoids build-phase `-Swift.h` dependency)

### 3. AvoEncryption.m Migration (core change)
- Removed `extern CCCryptorGCMOneshotEncrypt` declaration
- Removed `aesGcmEncrypt:` method calling private API
- Removed custom big-number functions (`computeYFromX:`, `bigMulMod:`, etc.)
- 12-byte nonce, version 0x01, 94-byte header
- NSLog gated on `[AvoInspector isLogging]`
- SHA-256 hash zeroed on stack after use

### 4. Tests
- `AvoEncryptionTests.m` — updated wire format constants (v0x01, 12-byte nonce offsets)
- All 29 encryption tests pass

## Known Limitations
- `AvoGCMEncryptor.swift` NSLog not gated on `isLogging` (Swift can't easily call ObjC class method)
- iOS 13-15 compressed key fallback relies on undocumented `SecKeyCreateWithData` behavior (low practical risk — server keys are uncompressed)

## Test plan
- [x] All 29 encryption tests pass with v0x01 wire format
- [x] Round-trip encrypt/decrypt verified
- [x] Wire format structure verified (version byte, nonce length, offsets)
- [x] Cross-platform verification with web app decryption (see `verification-report.md`)
- [x] Manual verification: `pod lib lint` passes
- [x] Manual verification: monorepo web app deployed with v0x01 support before SDK release

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Enhanced encryption implementation with an updated cryptographic protocol format for improved security.

* **Chores**
  * Bumped version to 3.0.0 (major release).
  * Increased minimum iOS deployment target to 13.0.
  * Added Swift 5.0 as a required language version.
  * Added Security and CryptoKit framework dependencies.

### CODE REVIEW

File: AvoInspector/Classes/AvoEncryption.m [hunk 1]
```diff
@@ -183,274 +172,37 @@ + (NSData * _Nullable)parseAndUncompressPublicKey:(NSData *)pubKeyBytes {
...
-    // Check parity
-    BOOL yIsOdd = (y[31] & 1) != 0;
-    if (yIsOdd != yOdd) {
-        uint8_t negY[32];
-        [self bigSubFromP:y p:p result:negY];
-        return [NSData dataWithBytes:negY length:32];
-    }
-
-    return [NSData dataWithBytes:y length:32];
+    return compressedKey;
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoEncryption.m]: Fixed in 4316561. On iOS 13-15 we now attempt a Security framework round-trip: SecKeyCreateWithData with the compressed key, then SecKeyCopyExternalRepresentation to get uncompressed bytes. If SecKeyCreateWithData rejects the compressed key, we return nil with a clear log message instead of silently passing unverified data.

File: AvoInspector/Classes/AvoEncryption.m [hunk 2]
```diff
@@ -503,7 +255,9 @@ + (NSData * _Nullable)computeECDHSharedSecret:(SecKeyRef)privateKey withPublicKe
 + (NSData *)sha256:(NSData *)data {
     uint8_t hash[CC_SHA256_DIGEST_LENGTH];
     CC_SHA256(data.bytes, (CC_LONG)data.length, hash);
-    return [NSData dataWithBytes:hash length:CC_SHA256_DIGEST_LENGTH];
+    NSData *hashData = [NSData dataWithBytes:hash length:CC_SHA256_DIGEST_LENGTH];
+    memset(hash, 0, CC_SHA256_DIGEST_LENGTH);
+    return hashData;
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoEncryption.m:278]: Fixed in af0c4dd. Replaced `memset` with `memset_s` which is guaranteed not to be optimized away by the compiler.

File: AvoInspector/Classes/AvoGCMEncryptor.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            let symKey = SymmetricKey(data: key as Data)
+            let nonce = try AES.GCM.Nonce(data: iv as Data)
+            let sealedBox = try AES.GCM.seal(plaintext as Data, using: symKey, nonce: nonce)
+            guard sealedBox.tag.count == 16 else { return false }
+            ciphertext.setData(Data(sealedBox.ciphertext))
+            authTag.setData(Data(sealedBox.tag))
+            return true
+        } catch {
+            NSLog("[avo] Avo Inspector: AES-GCM encryption failed: \(error)")
+            return false
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoGCMEncryptor.swift:18]: Acknowledged, leaving as-is. `AvoInspector.isLogging` is an ObjC class method not easily callable from Swift without importing the generated `-Swift.h` bridge header, which would create a circular dependency. The NSLog only fires on encryption failure (exceptional path) and the message is generic — no sensitive data is leaked.

File: Example/Tests/AvoEncryptionTests.m [hunk 4]
```diff
@@ -60,22 +60,22 @@ + (NSString *)publicKeyHexFromPrivateKey:(SecKeyRef)privateKey {
 
 + (NSString * _Nullable)decrypt:(NSString *)base64Encrypted privateKey:(SecKeyRef)privateKey {
     NSData *data = [[NSData alloc] initWithBase64EncodedString:base64Encrypted options:0];
-    if (data == nil || data.length < 98) return nil;
+    if (data == nil || data.length < 95) return nil;
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply Example/Tests/AvoEncryptionTests.m:63]: Fixed in af0c4dd. Changed minimum from 95 to 94 bytes (empty plaintext is a valid edge case).

File: planning/ecies-v1-wire-format/generate-fixture.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+//
+// Usage: node generate-fixture.js
+//
+// v0x01 wire format: [version(1)] [ephemeralPublicKey(65, uncompressed)] [nonce(12)] [authTag(16)] [ciphertext]
+// v0x00 wire format: [version(1)] [ephemeralPublicKey(65, uncompressed)] [iv(16)] [authTag(16)] [ciphertext]
+//
+// Note: iOS Security.framework always produces 65-byte uncompressed ephemeral public keys.
+// v0x01 uses 65-byte uncompressed keys to match the actual iOS SDK wire format.
+
+import crypto from "crypto";
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply planning/ecies-v1-wire-format/generate-fixture.js]: Moot — file removed in af0c4dd (planning artifacts removed from the PR).

File: planning/ecies-v1-wire-format/verification-report.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+---
+
+## Detailed Results
+
+### Item 1: CCCryptorGCMOneshotEncrypt scan
+
+**Command:**
+```
+grep -r CCCryptorGCMOneshotEncrypt AvoInspector/Classes/
+```
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply planning/ecies-v1-wire-format/verification-report.md]: Moot — file removed in af0c4dd (planning artifacts removed from the PR).

File: planning/ecies-v1-wire-format/verification-report.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+**Result: PASS**
+
+Output: `AvoInspector passed validation.` (exit code 0)
+
+Only informational `NOTE` entries were emitted (codesigning identity override, build order notes, metadata extraction skipped). No warnings or errors.
+
+**Note on binary symbol check:** Full binary symbol verification via `nm -u` (to confirm `CCCryptorGCMOneshotEncrypt` is absent from the linked binary) requires a complete Xcode build and archive. This is a **manual/CI gate** that must be performed before tagging the v3.0.0 release. The pod lint validates source and build but does not produce a final release binary for `nm` inspection. CI should run `nm -u` against the compiled `.framework` or `.a` artifact and confirm zero occurrences of `CCCryptorGCMOneshotEncrypt`.
+
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply planning/ecies-v1-wire-format/verification-report.md]: Moot — file removed in af0c4dd (planning artifacts removed from the PR).

**Review by logason — APPROVED** [review]
